src/platforms/linux/daemon/wireguardutilslinux.cpp

```diff
@@ -677,18 +689,66 @@ bool WireguardUtilsLinux::setupCgroupClass(const QString& path,
   return true;
 }
 
-QString WireguardUtilsLinux::getExcludeCgroup() const {
-  if (m_cgroups.isNull()) {
-    return QString();
+// static
+bool WireguardUtilsLinux::moveCgroupProcs(const QString& src,
+                                          const QString& dest) {
+  QFile srcProcs(src + "/cgroup.procs");
+  FILE* fp = fopen(qPrintable(dest + "/cgroup.procs"), "w");
+  if (!fp) {
```

Comment:
why not a QFile?

Reply:
It turns out that when writing to the cgroupsfs pseudo-filesystem, the kernel is sensitive to system call, and you can only write a single PID per `write()` call. Unfortunately a `QFile` tends to batch up the file data and submits it to the kernel in a single system call - resulting in some processes failing to migrate between control groups.